specs/approved/whole-system-configuration-steps.rst

```diff
@@ -4,85 +4,151 @@
+  via fewer required reboots, which is dependent on system and ironic hardware
   type support
 
-
-This specification proposes a generally applicable solution as new clean and
-deploy steps that each vendor hardware type can implement according to its
-capabilities.
+This specification proposes a generally applicable solution comprised of new
+cleaning and deploy steps which process system configurations. Each hardware
+type may implement it according to its and its system's capabilities.
 
 
 Problem description
 ===================
 
-When there are dozens of servers that need the same configuration, it is still
-a repetitive task to configure them all. Use cases to benefit from this
-include:
+A large number of systems with identical or similar hardware arrives. The user
+wants all of their hardware configured the same.
 
-* Consistent, rapid, and correct configuration of fleets of bare metal systems
-  with similar hardware capabilities during deployment
-* Production environments in which physical servers are repurposed for
-  workloads that require different configurations and fast turnaround is at
-  a premium
-* Getting inventory of system after changes are applied
+* In ironic, that can be a tedious, error-prone task.
+* Ironic is not capable of configuring all of the hardware subsystems for which
+  vendors offer configurability.
+* Applying a configuration can take a great deal of time, often requiring
+  multiple reboots.
 
+The same challenges apply when different workflows are run on the same system,
+each needing a different hardware configuration.
 
 Proposed change
 ===============
 
-This proposes adding new clean and deploy steps to configure a system as a
-single step and reuse configuration from already configured system with similar
-capabilities. This is proposing alternative, not replacement to currently
-existing granular configuration steps like RAID and BIOS configuration. More on
-how this compares to currently available functionality in Ironic is described
-in section `Alternatives`_.
-Overview is described at the beginning of section `Solution`_.
-A step to get current configuration from a system and store it in
-configuration mold is described in `Export configuration`_ section.
-A step to apply already stored configuration mold to a system is described in
-`Import configuration`_ section.
-A step that combines both operations into single step is described in section
-`Import and Export configuration`_.
-The data format that is used for configuration molds is described in section
-`Format of configuration data`_.
-How Ironic and its users interact with configuration molds is described in
-section `Storage of configuration data`_.
+Introduction
+------------
+
+The following concept is used in this specification:
+
+**configuration mold**
+    A document which expresses the current or desired configuraton of a
+    hardware system. It also offers an inventory of a system's hardware
+    configuration.
```

Comment:
I initially tried to define _configuration mold_ later in this subsection, immediately after the descriptions of the three (3) pairs of steps. In that iteration, the descriptions did not refer to that concept. Unfortunately, the formatting of that in the middle of the bullet list did not look good to me. So, I decided to move the definition to the top.

The downside is that the term is defined before the steps are described. The upside is the descriptions refer to the term. Since the definition here is quite brief and greater detail about it is provided after the descriptions of the steps, it seems to work.

What do you think?

Reply:
looks good